frontend/components/ui/FavoriteButton/favorite-button.component.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useCallback } from "react"
+import { useSelector } from "react-redux"
+import cx from "classnames"
+import { useApi } from "../../../hooks/useApi"
+import { IStoreState } from "../../../redux/store"
+import { IFullBuild, IUser } from "../../../types/models"
+import * as SC from "./favorite-button.styles"
+
+interface IFavoriteButtonProps
+  extends React.ComponentPropsWithoutRef<"button"> {
+  owner: IFullBuild["owner"]["username"]
+  slug: IFullBuild["slug"]
+}
+
+const FavoriteButton: React.FC<IFavoriteButtonProps> = ({ owner, slug }) => {
```

Comment:
I'd like to avoid getting the followers on every build view. Instead, we should show only the **count** and hide the full list behind an action. That's what GitHub does for example.

Instead of the custom logic to determine `isFollower`, you can just check for existence of the [`add_favorite` and `remove_favorite` links](https://github.com/factorio-builds/factorio-builds-tech/blob/main/frontend/types/generated-api.ts#L1085-L1096). That way you also don't have to hard code the url and can just use whatever the API is giving you.

Reply:
Ah right! That felt off for me! I'll change that

Reply:
currently there is no way for me to get the new updated count unless I either:

- refetch the build
- refetch the GET followers endpoint
- manually add/subtract 1 from initial count of the component

Reply:
updated

Reply:
It's not that big of a deal, now the action of favoriting/unfavouriting gets us the latest count. Given how unfrequent this action is, I didn't see it as an issue, performance/load-wise.

Not that the latest count is a critical information. Would you prefer I implement number 3 instead?

Reply:
I think either is fine; you're right that the action is not going to be frequent. The important thing is not to get the collection with every page view. Generally speaking though, in cases like these imho the pattern should be that you just fudge the count on the client side. Sure, it might end up being off by one or two if somebody else happens to fav at the same time, but I think that's an acceptable tradeoff for minimizing calls. From a UX perspective, I think you'd want to inrease the count instantly anyway and execute the call in the background. That way the action feels instant even if the user has a slow connection.